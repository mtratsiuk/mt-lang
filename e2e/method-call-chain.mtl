(def createList
  (|list l r|
    (cond
      ((= l r) list)
      (else (createList (list/concat l) (+ l 1) r)))))

(def testSum
  (~> ["1" "2" "3"]
   ~> map Number
   ~> reduce (|c r| (+ r c)) 0))

(def testDynMethod
  (~> ["1" "2" "3"]
   ~> (~> [".m." ".a." ".p."] ~> join "" ~> replaceAll "." "") Number
   ~> reduce (|c r| (+ r c)) 0))

(def testDynReceiver
  (~> (createList [] 0 10)
   ~> filter (|x| (= (% x 2) 0))
   ~> join " | "))

(print testSum)
(print testDynMethod)
(print testDynReceiver)
