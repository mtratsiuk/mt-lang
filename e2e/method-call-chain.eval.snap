6
6
0 | 2 | 4 | 6 | 8
